test/test_using_plams/test_cp2k_mm.py

```diff
@@ -0,0 +1,176 @@
+from os import PathLike
+from io import TextIOBase
+from itertools import islice
+from typing import Union, AnyStr, Any
+from collections import abc
+
+import numpy as np
+import pytest
+from assertionlib import assertion
+from scm.plams import Molecule, add_to_instance, add_to_class, Units, Cp2kJob
+
+from qmflows import Settings, run, cp2k_mm, singlepoint, geometry, freq, md
+from qmflows.backports import nullcontext
+from qmflows.test_utils import delete_output, get_mm_settings, PATH
+
+MOL = Molecule(PATH / 'Cd68Cl26Se55__26_acetate.xyz')
+SETTINGS: Settings = get_mm_settings()
+
+
+def get_frequencies(file: Union[AnyStr, PathLike, TextIOBase],
```

Comment:
Aren't these constraints very loose? meaning that maybe you can enforce the client code to call this function with anything that is file/iterable, instead of this function having to handle those corner cases

Reply:
This function should most definitely be moved somewhere else (probably the ``Cp2kResults`` class).
For now I simply wanted a way to extract the vibrational frequencies.

As for the constraints, these more or less represent all path- and file-like objects.
``open()`` can handle the former and the latter can simply be handler via iteration.
And they're more like guidelines rather than strict rules anyway.